pkg/master/thirdparty/thirdparty.go

```diff
@@ -284,6 +284,7 @@ func (m *ThirdPartyResourceServer) InstallThirdPartyResource(rsrc *extensions.Th
 	m.genericAPIServer.HandlerContainer.Add(genericapi.NewGroupWebService(api.Codecs, path, apiGroup))
 
 	m.addThirdPartyResourceStorage(path, plural.Resource, thirdparty.Storage[plural.Resource].(*thirdpartyresourcedataetcd.REST), apiGroup)
+	registered.AddThirdPartyAPIGroupVersions(schema.GroupVersion{Group: group, Version: rsrc.Versions[0].Name})
```

Comment:
@sttts along with your changes to add verbs for thirdpartyresources, the namespace controller will be aware of them, so let's reconsider the problem at https://github.com/kubernetes/kubernetes/issues/37278.

When a namespace is going to be deleted, the namespace controller will delete all the deletable resources under that namespace, and in case it is a TPR resources, the delete options will failed to be decoded by a TPR gvk, and @caesarxuchao made a change in this pull request: https://github.com/kubernetes/kubernetes/pull/37328, if you look at the code, it checks whether the gvk is a TPR group version by calling `registered.IsThirdPartyAPIGroupVersion`, and that call depends on whether we register it or not.